src/backend-api/Controllers/AddController.cs

```
@@ -518,6 +524,53 @@ public IActionResult PostProgram([FromBody] PostProgramInputModel input)
                 return BadRequest();
             }
         }
+        /* POST: api/add/Plugin
+         * Takes in as input:
+         * {
+	     *     "ProgramName" : String,
+	     *     "PluginName" : int,
+	     *     "PluginFlatCost" : Decimal,
+	     *     "TextField" : string,
+	     *     "PluginCostPerYear" : Decimal,
+	     *     "RenewalDate" : DateTime,
+	     *     "MonthsPerRenewal" : int,
+         *     "DateBought" : DateTime
+         * }
+         */
+
+        [HttpPost]
+        [Route("Plugin")]
+        public IActionResult PostPlugin([FromBody] PostPluginInputModel input)
+        {
+            if (!(_context.Program.Select(x => x.ProgramName).ToList().Contains(input.ProgramName)))
+                return BadRequest("No such program exists");
+            try
+            {
+                var plugin = new Plugins()
+                {
+                    PluginName = input.PluginName,
+                    PluginFlatCost = input.PluginFlatCost,
+                    ProgramId = _context.Program.Where(x => x.ProgramName == input.ProgramName).Select(x => x.ProgramId).First(),
+                    TextField = input.TextField,
+                    PluginCostPerYear = input.PLuginCostPerYear,
+                    IsDeleted = false,
+                    ProgramName = input.ProgramName,
+                    RenewalDate = input.RenewalDate,
+                    MonthsPerRenewal = input.MonthsPerRenewal,
+                    Datebought = input.DateBought,
+                    IsCostPerYear = input.MonthsPerRenewal != null && input.MonthsPerRenewal - 12 >= 0 ? true : false,
+                };
+                _context.Add(plugin);
+                _context.SaveChanges();
+                return StatusCode(201);
+            }
+            catch
+            {
+                return BadRequest();
```

From a maintainability standpoint, I'd prefer that this `try/catch/return BadRequest()` pattern be omitted so that the exception can propagate. When you do this, it removes contextual information from the error so you have no idea why something failed. If you instead let the exception be thrown, it will be propagated either to the client or some general catch-all-and-log-errors middleware.

---

Alrighty, that makes sense